src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -714,17 +714,31 @@ def get_validation_result(self, use_scratch_stage: bool):
                         f"drop stage if exists {self.scratch_stage_fqn}"
                     )
 
-    def get_events(self) -> list[dict]:
+    def get_events(
+        self, since_interval: str = "", until_interval: str = ""
+    ) -> list[dict]:
         if not self.account_event_table:
             raise NoEventTableForAccount()
 
         # resource_attributes:"snow.database.name" uses the unquoted/uppercase app name
         app_name = unquote_identifier(self.app_name)
+        since_clause = (
+            f"and timestamp >= sysdate() - interval '{since_interval}'"
+            if since_interval
+            else ""
+        )
+        until_clause = (
+            f"and timestamp <= sysdate() - interval '{until_interval}'"
+            if until_interval
+            else ""
+        )
```

Comment:
this could also support absolute timestamps in the future, I just wanted a quick and easy way to continue developing and testing but avoid fetching days worth of events by just doing `snow app events --since '1 hour'`